### PR TITLE
#421 Cannot create subclass issue

### PR DESCRIPTION
**Pull request by naveen2507**
- Issue #421 
- Issue because of NO subClassOf property found in KB. 

Solution Description: 

- Initialize base properties while registering or updating KB
- Base Properties : subClassOf, comment  and label 
- When updating KB , delete the list of older base properties.

### CODE REVIEW

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java [hunk 1]
```diff
@@ -355,4 +355,12 @@ RepositoryImplConfig getKnowledgeBaseConfig(KnowledgeBase kb)
     List<KBQualifier> listQualifiers(KnowledgeBase kb, KBStatement aStatement);
 
     boolean statementsMatchSPO(KnowledgeBase akb, KBStatement mockStatement);
+    
+    /**
+     * Define base properties of comment, label and subClassOf while initializing the KB
+     * @param kb The knowledge base to initialize base properties
+     * @param kbUpdateFlag Flag variable to indicate the update of KB or not.  
+     */
+    void defineBaseProperties(KnowledgeBase kb, boolean kbUpdateFlag);
```
Comment (Rentier) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java]: Why do you pass the `kbUpdateFlag`? `KnowledgeBase` has a field `readOnly`

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java [hunk 2]
```diff
@@ -355,4 +355,12 @@ RepositoryImplConfig getKnowledgeBaseConfig(KnowledgeBase kb)
     List<KBQualifier> listQualifiers(KnowledgeBase kb, KBStatement aStatement);
 
     boolean statementsMatchSPO(KnowledgeBase akb, KBStatement mockStatement);
+    
+    /**
+     * Define base properties of comment, label and subClassOf while initializing the KB
+     * @param kb The knowledge base to initialize base properties
+     * @param kbUpdateFlag Flag variable to indicate the update of KB or not.
```
Comment (Rentier) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java]: Can you elaborate this a bit? I do not get it quite.

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java [hunk 3]
```diff
@@ -158,6 +158,36 @@ public void registerKnowledgeBase(KnowledgeBase kb, RepositoryImplConfig cfg)
...
+    {
+        if (kbUpdateFlag) {
+            List<KBHandle> listProperties = listProperties(kb, true);
+            if (!listProperties.isEmpty()) {
+                for (KBHandle propertyHandle : listProperties) {
+                    KBProperty property = new KBProperty();
+                    property.setIdentifier(propertyHandle.getIdentifier());
+                    property.setName(propertyHandle.getName());
+                    deleteProperty(kb, property);
+
```
Comment (Rentier) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java]: Whitespace

File: inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImplIntegrationTest.java [hunk 4]
```diff
@@ -311,6 +311,14 @@ public void empty_WithNonemptyKnowledgeBase_ShouldReturnFalse() {
             .isFalse();
     }
 
+    @Test
+    public void nonempty_WithEmptyKnowledgeBase_ShouldReturnTrue() {
+        sut.registerKnowledgeBase(kb, sut.getNativeConfig());
+        sut.defineBaseProperties(kb,false);
+        List<KBHandle> listProperties = sut.listProperties(kb, true);
+        assertEquals(listProperties.size(),3);
```
Comment (Rentier) [inline inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImplIntegrationTest.java]: Please also assert that the list contains the IRIs you expect, not only the size.

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java (hunk 1 above)
Comment (naveen2507) [reply inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseService.java]: Its a new feature in there , where we create a default list of properties and when we update the set of schema values in there, we need this flag.

**Pull request merged**